web/web.go

```diff
@@ -110,6 +111,40 @@ func (s *PrometheusStatus) ApplyConfig(conf *config.Config) bool {
 	return true
 }
 
+// PrometheusVersion contains various build information about Prometheus
+type PrometheusVersion struct {
+	Version   string
+	Revision  string
+	Branch    string
+	BuildUser string
+	BuildDate string
+	GoVersion string
+}
+
+// Map provides the iterable version information.
+func (pv *PrometheusVersion) Map() map[string]interface{} {
```

Comment:
This looks overkill to me. Let's just change the template to use the keys explicitly and remove the iteration.